src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,52 +40,1012 @@ const ActorInit En_Niw_InitVars = {
+    f32 dX;
+    f32 dZ;
+
+    // it actually wants to copy to stack... not modify, then pass to veccopy
+    // and it does it BEFORE the if block, this is just unoptimized.
+    tempVec3f = D_808934E8;
+    if (this->unkTimer254 == 0) {
+        // the assignements in this block never want to align properly
+        // its not lineswap, as 800k permuter with only lineswap didn't find a solution
+        this->unk298 = 0;
+
+        this->unk2A4.x = this->actor.world.pos.x;
+        this->unk2A4.y = this->actor.world.pos.y;
+        this->unk2A4.z = this->actor.world.pos.z;
+        this->unk2B0.x = this->actor.world.pos.x;
+        this->unk2B0.y = this->actor.world.pos.y;
+        this->unk2B0.z = this->actor.world.pos.z;
+
+        this->unkTimer252 = this->unkTimer250 = this->unk298;
+        this->unk264[8] = 0;
+        this->unk264[6] = 0;
+        this->unk264[5] = 0;
+        this->unk264[7] = 0;
+        this->unk304 = 0;
+        this->unk300 = 0;
+        this->actor.speedXZ = 0;
+        Math_Vec3f_Copy(&this->unk2BC, &tempVec3f);
+        EnNiw_SetupIdle(&this->actor);
+    } else {
+        if (this->unk2BC.x != 90000.0f) {
+            dX = this->actor.world.pos.x - this->unk2BC.x;
+            dZ = this->actor.world.pos.z - this->unk2BC.z;
+        } else {
+            dX = this->actor.world.pos.x - player->base.world.pos.x;
+            dZ = this->actor.world.pos.z - player->base.world.pos.z;
+        }
+        this->yawTowardsPlayer = Math_Atan2S(dX, dZ);
+        func_808917F8(this, globalCtx, 0);
+        func_80891320(this, globalCtx, 2);
+    }
+}
+#else
 #pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_808924B0.asm")
+#endif
+
+// check if on the ground after running, once on the ground, start idling
+void EnNiw_LandBeforeIdle(EnNiw* this, GlobalContext* gCtx) {
+    if ((this->actor.bgCheckFlags & 1)) { // hit floor
+        EnNiw_SetupIdle(this);
+    }
+}
+
+void EnNiw_CheckRage(EnNiw* this, GlobalContext* globalCtx) {
+    if ((!this->niwStormActive) && (this->unkTimer260 == 0) && (this->niwType == ENNIW_TYPE_REGULAR)) {
+        if ((this->unknownState28E != 7) && (90000.0f != this->unk2BC.x)) {
+            this->unkTimer260 = 10;
+            this->sfxTimer1 = 30;
+            this->unk29E = 1;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M); // crow
+            this->unkTimer254 = 100;
+            this->unk2EC = 0;
+            EnNiw_SetupRunning(this);
+        }
+
+        if ((this->collider.base.acFlags & AC_HIT)) {
+            this->collider.base.acFlags &= ~AC_HIT;
+            // health gets used as a hit counter until cucco storm
+            if (this->actor.colChkInfo.health > 0) {
+                this->actor.colChkInfo.health--;
+            }
+
+            if ((D_80893460 == 0) && (this->actor.colChkInfo.health == 0)) {
+                // now you've done it
+                this->unkTimer254 = 100;
+                D_80893460 = 1;
```

Comment:
```suggestion
            if (!D_80893460 && (this->actor.colChkInfo.health == 0)) {
                // now you've done it
                this->unkTimer254 = 100;
                D_80893460 = true;
```
Looks like `D_80893460` is boolean